### PR TITLE
Feat: added the API that we can verify attendee token

### PR DESCRIPTION
**Pull request by kaka-lin**
## Types of changes

- [ ] **Bugfix**
- [x] **New feature**
- [ ] **Refactoring**
- [ ] **Breaking change** (any change that would cause existing functionality to not work as expected)
- [ ] **Documentation Update**
- [ ] **Other (please describe)**

## Description

The API that we can verify attendee token, and return youtube's information

## Related Issue

 [Revamp `ext2020` app](https://trello.com/c/vYNToMLT/50-revamp-ext2020-app)

### CODE REVIEW

**Comment by kaka-lin** [comment]
[Revamp `ext2020` app](https://trello.com/c/vYNToMLT/50-revamp-ext2020-app)

File: src/ext2020/api/views.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from collections import OrderedDict
+
+from django.conf import settings
+from registry.helper import reg
+from rest_framework import views
+from rest_framework.response import Response
+from rest_framework import status
```
Comment (mattwang44) [inline src/ext2020/api/views.py:7]:
```suggestion
from rest_framework import views, status
from rest_framework.response import Response
```

File: src/ext2020/api/views.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from collections import OrderedDict
```
Comment (mattwang44) [inline src/ext2020/api/views.py:1]: unused

File: src/ext2020/api/views.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class AttendeeAPIView(views.APIView):
+    model = Attendee
+
+    def post(self, request):
+        data = []
+        self.token = request.data['token']
+        try:
+            Attendee.objects.get(token=self.token)
+            response_data = {"youtube_infos": []}
+            for i in range(1, 4):
```
Comment (mattwang44) [inline src/ext2020/api/views.py:21]: how to confirmed there are 4 rooms?

**Comment by uranusjr** [comment]
Can we put the code into `ext2021` instead? It’s no longer 2020…

**Comment by mattwang44** [comment]
> Can we put the code into `ext2021` instead? It’s no longer 2020…

Yes we planned to do this (see the attached Trello ticket). This PR is still on its halfway.

**Comment by mattwang44** [comment]
[Revamp `ext2020` app](https://trello.com/c/vYNToMLT/50-revamp-ext2020-app)

File: src/attendee/api/views.py [hunk 4]
```diff
@@ -1,9 +1,8 @@
 from django.conf import settings
-from registry.helper import reg
-from rest_framework import views
+from rest_framework import views, status
 from rest_framework.response import Response
-from rest_framework import status
 
+from registry.helper import reg
 from ext2020.models import Attendee
```
Comment (mattwang44) [inline src/attendee/api/views.py]: sort the imports by:

1. native packages
2. 3rd party libraries
3. customized packages

and use an empty line to separate them.

```suggestion
from registry.helper import reg

from ext2020.models import Attendee
```

File: src/attendee/api/views.py [hunk 5]
```diff
@@ -14,14 +13,15 @@ def post(self, request):
         self.token = request.data['token']
         try:
             Attendee.objects.get(token=self.token)
+            slug = settings.CONFERENCE_DEFAULT_SLUG
+            key_filter = f"{slug}.live"
+            num_rooms = [key for key in reg if key_filter in str(key)]
```
Comment (mattwang44) [inline src/attendee/api/views.py:18]:
```suggestion
            key_prefix = f"{settings.CONFERENCE_DEFAULT_SLUG}.live"
            reg_room_keys = [key for key in reg if str(key).startswith(key_prefix)]
```

File: src/attendee/api/views.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+from django.conf import settings
+from rest_framework import views, status
+from rest_framework.response import Response
+
+from registry.helper import reg
+from ext2020.models import Attendee
```
Comment (mattwang44) [inline src/attendee/api/views.py:6]: sort the imports by:

1. native packages
2. 3rd party libraries
3. customized packages

and use an empty line to separate them.

```suggestion
from registry.helper import reg

from ext2020.models import Attendee
```

File: src/attendee/api/views.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+            slug = settings.CONFERENCE_DEFAULT_SLUG
+            key_filter = f"{slug}.live"
+            num_rooms = [key for key in reg if key_filter in str(key)]
+
+            response_data = {"youtube_infos": []}
+            for idx, room in enumerate(num_rooms):
+                response_data["youtube_infos"].append({
+                    "room": f"R{idx+1}",
+                    "video_id": reg.get(f'{room}', '')
+                })
```
Comment (mattwang44) [inline src/attendee/api/views.py:25]: Let's take the last part of the registry key as the room name. The following suggestion is not tested:

```suggestion
            for idx, room_key in enumerate(reg_room_keys):
                pattern = r'{slug}\.live\.(?P<room_name>[0-9a-zA-Z]*)'
                match_result = re.match(pattern, room_key)
                if not match_result:
                    continue
                room_name = match_result.group(1)
                
                response_data["rooms"].append({
                    "room": room_name,
                    "video_id": reg.get(room_key, '')
                })
```

File: src/attendee/api/views.py [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    def post(self, request):
+        self.token = request.data['token']
+        try:
+            Attendee.objects.get(token=self.token)
+            slug = settings.CONFERENCE_DEFAULT_SLUG
+            key_filter = f"{slug}.live"
+            num_rooms = [key for key in reg if key_filter in str(key)]
+
+            response_data = {"youtube_infos": []}
+            for idx, room in enumerate(num_rooms):
```
Comment (mattwang44) [inline src/attendee/api/views.py:21]: Consider using the room name defined in `src/events/models.py > Location`.

File: src/attendee/api/views.py (hunk 6 above)
Comment (mattwang44) [reply src/attendee/api/views.py:6]: `ext2020` is going to be deprecated. Please remove the dependencies on this app.

**Review by mattwang44 — APPROVED** [review]
LGTM. There are still two tasks to be done. Let's deal with them in the following PRs. 

1. remove `ext2020`.
2. refine the logic to return the room name in response body.

**Pull request merged**